### ISSUE TITLE
Batch normalisation irroneous

### ISSUE THREAD

**Issue by WillRNaylor** [arangIN/SelfOrganizingMap-SOM#4]
I have been using your code (thank you very much for the nice repo), and I wonder if line 85 of som.py should be removed. The line in question simply has:

`delta.div_(batch_size)`

But I suspect this shouldn't be done, since SOM training isn't a batch gradient descent or anything - you are using batches just for computational efficiency, and simply adding all the updates and doing them at the same time.

I could be wrong though.

And thanks again for the lovely repo.

**Comment by Dotori-HJ** [comment]
As you say, SOM update itself one by one. However, I found results to change more smoothly when updating to the average of several delta values ​​with the batch size. So I added the batch size to the code. In addition, I just added the batch size for my study. I am contemplating this and studying in detail.

If you want to use the original version of SOM, you can set batch_size=1.

Thanks for your interest in my repo.
